pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1054,23 +1057,30 @@ func getTaskrunWorkspaces(ctx context.Context, pr *v1beta1.PipelineRun, rpt *res
 
 // taskWorkspaceByWorkspaceVolumeSource is returning the WorkspaceBinding with the TaskRun specified name.
 // If the volume source is a volumeClaimTemplate, the template is applied and passed to TaskRun as a persistentVolumeClaim
-func taskWorkspaceByWorkspaceVolumeSource(wb v1beta1.WorkspaceBinding, taskWorkspaceName string, pipelineTaskSubPath string, owner metav1.OwnerReference) v1beta1.WorkspaceBinding {
+func (c *Reconciler) taskWorkspaceByWorkspaceVolumeSource(ctx context.Context, pipelineWorkspaceName string, prName string, wb v1beta1.WorkspaceBinding, taskWorkspaceName string, pipelineTaskSubPath string, owner metav1.OwnerReference) (v1beta1.WorkspaceBinding, error) {
 	if wb.VolumeClaimTemplate == nil {
 		binding := *wb.DeepCopy()
 		binding.Name = taskWorkspaceName
 		binding.SubPath = combinedSubPath(wb.SubPath, pipelineTaskSubPath)
-		return binding
+		return binding, nil
 	}
 
-	// apply template
+	claim := corev1.PersistentVolumeClaim{}
 	binding := v1beta1.WorkspaceBinding{
-		SubPath: combinedSubPath(wb.SubPath, pipelineTaskSubPath),
-		PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-			ClaimName: volumeclaim.GetPersistentVolumeClaimName(wb.VolumeClaimTemplate, wb, owner),
-		},
+		SubPath:               combinedSubPath(wb.SubPath, pipelineTaskSubPath),
+		PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{},
 	}
 	binding.Name = taskWorkspaceName
-	return binding
+
+	pvcName := volumeclaim.GetPersistentVolumeClaimName(&claim, wb, owner)
```

Comment:
does the function `GetPersistentVolumeClaimName` need to be updated? This is not the only place this function is used.

Reply:
`GetPersistentVolumeClaimName` does not need to be updated as we are not changing the pvc naming logic

Reply:
Can you double check? If we're applying the `volumeClaimTemplate` to the statefulset spec, I believe the PVCs generated will have different names than they did previously.

Reply:
I agree it is confusing! `GetPersistentVolumeClaimName` actually refers to `GetPersistentVolumeClaimNameWithoutAffinityAssistant` so we don't need to update this part of logic. I have update the function name and string doc to improve readability, PTAL!